packages/core/src/render3/node_manipulation.ts

```diff
@@ -125,32 +102,45 @@ export function createTextNode(value: string, renderer: Renderer3): RText {
 }
 
 /**
- * Adds or removes all DOM elements associated with a view.
+ * Removes all DOM elements associated with a view.
  *
  * Because some root nodes of the view may be containers, we sometimes need
  * to propagate deeply into the nested containers to remove all elements in the
  * views beneath it.
  *
  * @param tView The `TView' of the `LView` from which elements should be added or removed
  * @param lView The view from which elements should be added or removed
- * @param insertMode Whether or not elements should be added (if false, removing)
+ */
+export function removeViewFromContainer(tView: TView, lView: LView): void {
+  const renderer = lView[RENDERER];
+  applyView(tView, lView, renderer, WalkTNodeTreeAction.Detach, null, null);
+  lView[HOST] = null;
+  lView[T_HOST] = null;
+}
+
+/**
+ * Adds all DOM elements associated with a view.
+ *
+ * Because some root nodes of the view may be containers, we sometimes need
+ * to propagate deeply into the nested containers to add all elements in the
+ * views beneath it.
+ *
+ * @param tView The `TView' of the `LView` from which elements should be added or removed
+ * @param renderTNode The `TNode` where the `LView` should be attached to.
+ * @param renderer Current renderer to use for DOM manipulations.
+ * @param lView The view from which elements should be added or removed
+ * @param renderParent The parent `RElement` where it should be inserted into.
  * @param beforeNode The node before which elements should be added, if insert mode
  */
-export function addRemoveViewFromContainer(
-    tView: TView, lView: LView, insertMode: true, beforeNode: RNode|null): void;
-export function addRemoveViewFromContainer(
-    tView: TView, lView: LView, insertMode: false, beforeNode: null): void;
-export function addRemoveViewFromContainer(
-    tView: TView, lView: LView, insertMode: boolean, beforeNode: RNode|null): void {
-  const renderParent = getContainerRenderParent(tView.node as TViewNode, lView);
-  ngDevMode && assertNodeType(tView.node as TNode, TNodeType.View);
-  if (renderParent) {
-    const renderer = lView[RENDERER];
-    const action = insertMode ? WalkTNodeTreeAction.Insert : WalkTNodeTreeAction.Detach;
-    applyView(tView, lView, renderer, action, renderParent, beforeNode);
-  }
+export function addViewToContainer(
+    tView: TView, renderTNode: TNode, renderer: Renderer3, lView: LView, renderParent: RElement,
```

Comment:
I'd propose renaming `renderTNode` -> `parentTNode` and may be `renderParent` -> `parentNativeNode` (so both argument names are in sync):
```suggestion
    tView: TView, lView: LView, renderer: Renderer3, parentTNode: TNode, parentNativeNode: RElement,
```